src/main/java/net/neoforged/moddevgradle/internal/ModDevPlugin.java

```diff
@@ -165,36 +165,41 @@ public void apply(Project project) {
             });
         });
 
-        // it has to contain client-extra to be loaded by FML, and it must be added to the legacy CP
-        var createArtifacts = tasks.register("createMinecraftArtifacts", CreateMinecraftArtifactsTask.class, task -> {
-            task.setGroup(INTERNAL_TASK_GROUP);
-            task.setDescription("Creates the NeoForge and Minecraft artifacts by invoking NFRT.");
-
+        // Configure common properties of NeoFormRuntimeEngineTask
+        Consumer<NeoFormRuntimeEngineTask> configureEngineTask = task -> {
             var nfrtSettings = extension.getNeoFormRuntime();
             task.getVerbose().set(nfrtSettings.getVerbose());
             task.getEnableCache().set(nfrtSettings.getEnableCache());
             task.getAnalyzeCacheMisses().set(nfrtSettings.getAnalyzeCacheMisses());
             task.getUseEclipseCompiler().set(nfrtSettings.getUseEclipseCompiler());
             task.getArtifactManifestFile().set(createManifest.get().getManifestFile());
             task.getNeoForgeArtifact().set(extension.getVersion().map(version -> "net.neoforged:neoforge:" + version));
+            task.getNeoFormArtifact().set(extension.getNeoFormVersion().map(version -> "net.neoforged:neoform:" + version + "@zip"));
+            task.getNeoFormRuntime().from(neoFormRuntimeConfig);
+        };
+
+        // it has to contain client-extra to be loaded by FML, and it must be added to the legacy CP
+        var createArtifacts = tasks.register("createMinecraftArtifacts", CreateMinecraftArtifactsTask.class, task -> {
+            task.setGroup(INTERNAL_TASK_GROUP);
+            task.setDescription("Creates the NeoForge and Minecraft artifacts by invoking NFRT.");
+
             task.getAccessTransformers().from(accessTransformers);
             task.getParchmentData().from(parchmentData);
-            task.getNeoFormRuntime().from(neoFormRuntimeConfig);
 
             var minecraftArtifactsDir = modDevBuildDir.map(dir -> dir.dir("artifacts"));
             task.getCompiledArtifact().set(minecraftArtifactsDir.map(dir -> dir.file("neoforge-minecraft-joined-local.jar")));
             task.getCompiledWithSourcesArtifact().set(minecraftArtifactsDir.map(dir -> dir.file("neoforge-minecraft-joined-local-merged.jar")));
             task.getSourcesArtifact().set(minecraftArtifactsDir.map(dir -> dir.file("neoforge-minecraft-joined-local-sources.jar")));
             task.getResourcesArtifact().set(minecraftArtifactsDir.map(dir -> dir.file("neoforge-minecraft-joined-local-resources-aka-client-extra.jar")));
+
+            configureEngineTask.accept(task);
         });
 
         var downloadAssets = tasks.register("downloadAssets", DownloadAssetsTask.class, task -> {
-            task.setGroup(TASK_GROUP);
+            task.setGroup(INTERNAL_TASK_GROUP);
```

Comment:
I intentionally put this in the normal task group, as some may want to download assets without running a client (ie to prepare an offline run)